src/extension.ts

```diff
@@ -46,6 +52,13 @@ function customPush(name: string, handler: (...args: any[]) => any, context: vsc
     context.subscriptions.push(menuCommand);
 }
 
+function customUriPush(name: string,openAIRepo: OpenAIRepository, context: vscode.ExtensionContext):void{
+    const command=vscode.commands.registerCommand(name, async (uri: vscode.Uri) => {
+        await createResponsiveWidgetFromDescription(openAIRepo,uri);
```

Comment:
@tanmoy27112000 I am trying to run this command but it's not working because uri is always undefined. 

Also, please consider passing the method as a callback so the function `customUriPush` stays generalised.